lib/active_model_serializers/adapter.rb

```diff
@@ -0,0 +1,93 @@
+module ActiveModelSerializers
+  module Adapter
+    UnknownAdapterError = Class.new(ArgumentError)
+    ADAPTER_MAP = {}
+    private_constant :ADAPTER_MAP if defined?(private_constant)
+    require 'active_model_serializers/adapter/fragment_cache'
+    require 'active_model_serializers/adapter/cached_serializer'
+
+    class << self # All methods are class functions
+      def new(*args)
+        fail ArgumentError, 'Adapters inherit from Adapter::Base.' \
+          "Adapter.new called with args: '#{args.inspect}', from" \
+          "'caller[0]'."
+      end
+
+      def configured_adapter
```

Comment:
I know the specs for this were pre-existing and taken from `ActiveModel::Serializer.adapter` but the specs are missing two cases:

1. the config is set as a `String` e.g. `"json_api"`
2. the config is set as a `Symbol` or `String` than can be `underscore'd` e.g. `:JsonApi`

Maybe need a follow up PR.